security/token-file-provider.1.rst

```diff
@@ -0,0 +1,175 @@
+NAME
+====
+token-file-provider – Generate Vault tokens for EdgeX services
+
+
+SYNOPSIS
+========
+token-file-provider [-h|-help] [-c|--confdir <confdir>] [-p|--profile <name>]
+
+
+DESCRIPTION
+===========
+token-file-provider generates per-service Vault tokens for EdgeX services
+so that they can make authenticated connections to Vault to retrieve
+application secrets.
+token-file-provider implements a generic secret seeding mechanism based
+on pre-created files and is designed for maximum portability.
+token-file-provider takes a configuration file that specifies the services
+for which tokens shall be generated and the Vault access policy
+that shall be applied to those tokens.
+token-file-provider assumes that there is some underlying protection mechanism
+that will be used to prevent EdgeX services from reading each other’s tokens.
+
+
+OPTIONS
+=======
+  \-h, \--help
+    Display help text
+
+  \-c, \--confdir <confdir>
+    Look in this directory for configuration.toml instead.
+
+  \-p, \--profile <name>
+    Indicate configuration profile other than default
+
+
+FILES
+=====
+
+configuration.toml
+------------------
+This file specifies the TCP/IP location of the Vault service
+and parameters used for Vault token generation.
+
+::
+
+  [SecretService]
+  Server = "localhost"
+  Port = 8200 
+
+  [TokenFileProvider]
+  PrivilegedTokenPath = /run/edgex/secrets/token-file-provider/vault-token.json
+  ConfigFile = token-config.json
+  OutputDir = /run/edgex/secrets/
+  OutputFilename = vault-token.json
+
+
+vault-token.json
+----------------
+This file contains a token use to authenticate to Vault.
+
+::
+
+  {
+    "auth": {
+      "client_token": "s.wOrq9dO9kzOcuvB06CMviJhZ"
+    }
+  }
+
+
+token-config.json
+-----------------
+This configuration file tells token-file-provider which tokens to generate.
+
+In order to avoid a directory full of `.hcl` files,
+this configuration file uses the JSON serialization of HCL,
+documented at https://github.com/hashicorp/hcl/blob/master/README.md.
+
+::
+
+  {
+    "service-name": {
+      "edgex-use-defaults": true,
+      "custom_policy": [
+        {
+          "path": {
+            "secret/edgex/pki/tls/edgex-kong": {
```

Comment:
s/kong/security-proxy ?

Reply:
Actually, this is where it was in Edinburgh.

Reply:
we can break Edinburgh changes so let's make it generic for Fuji

Reply:
Ideally we won't have a special place for it, so how about, as an example, `secret/non/standard/location`?

Reply:
Applied above fix.